### ISSUE TITLE
DSPDFViewer : Both windows of presentation spawning on the same screen

### ISSUE THREAD

**Issue by alamasad84** [dannyedel/dspdfviewer#87]
I was trying to use dspdfviewer on windows for presentation made using LaTex. I have placed second screen on right. It opens two different windows, one for the presentation and one for notes. i have also extended the displays with my laptop screen on the right and projector screen on the left. However, both screens appear on my right screen, i.e., my laptop screen and i cannot move the main presentation to the projector screen. Important details are
1) OS : Windows7
2) Version: v1.13-218-g83eb2d8
3) Installed from @projekter's setup.exe

**Comment by projekter** [comment]
I observe the same thing whenever I open the presentation; simply hit 's' and the presentation appears on the correct screens. I never thought of reporting it, but you're right, this should be fixed.

**Comment by alamasad84** [comment]
Yes, it works. thanks.

On Tue, Nov 3, 2015 at 3:08 PM, Benjamin Desef <notifications@github.com>
wrote:

> I observe the same thing whenever I open the presentation; simply hit 's'
> and the presentation appears on the correct screens. I never thought of
> reporting it, but you're right, this should be fixed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dannyedel/dspdfviewer/issues/87#issuecomment-153366437>
> .
>



-- 
Syed Asad Alam
PhD Student
Computer Engineering, ISY
Linköping University,
Linköping, Sweden
alam.asad [at] gmail [dot] com
syed [dot] asad [dot] alam [at] liu [dot] se
Office: 0046-(0)13 28 13 57
Cell : 0046-(0)76 582 81 00
Personal Website <http://sites.google.com/site/alamasad/>
Official Website <http://www.es.isy.liu.se/staff/asad/index.html>
Monster Profile <http://beknown.com/syed.asad.alam>

**Comment by projekter** [comment]
Ok, I've looked into it. The problem is in PDFViewerWindow::reposition().
There are basically two possibilities:
- Remove all the rect retrieval and position setting stuff and replace it by a call of windowHandle()->setScreen(), followed by showFullScreen(). I think that this is the cleaner approach, as it really tells the OS that this window is maximized on a different screen. Or
- First call showFullScreen() - this will always go for the first screen - and the reposition the window. The reverse order worked in Qt4, but [does no longer in Qt5](http://stackoverflow.com/questions/3203095/display-window-full-screen-on-secondary-monitor-using-qt). The workaround that I presented above (hitting 's') only works because of showFullScreen essentially does nothing when called the second time, if the window style did not change in any way.

I tried both possibilities; swapping is always slow when triggered for the first time (but the screens are correctly positioned from the beginning). No idea why - maybe because of the differing screen resolutions? But the second option always causes the taskbar to appear in front of the secondary window (and possibly the audience window in Windows 10 where you have a taskbar on every screen). So I will create a pull request which realizes the first option. I guess it should be compatible with all other OSes as well.
